src/xenia/kernel/xam/content_manager.cc

```diff
@@ -259,6 +264,29 @@ std::filesystem::path ContentManager::ResolveGameUserContentPath() {
   return root_path_ / title_id / kGameUserContentDirName / user_name;
 }
 
+const std::vector<object_ref<XFile>> ContentManager::GetOpenedFilesFromContent(
+    const std::string_view root_name) {
+  std::vector<object_ref<XFile>> opened_files = {};
+  const std::vector<object_ref<XFile>> all_files_handles =
+      kernel_state_->object_table()->GetObjectsByType<XFile>(
+          XObject::Type::File);
+
+  std::string resolved_path = "";
+  kernel_state_->file_system()->FindSymbolicLink(std::string(root_name) + ':',
+                                                 resolved_path);
+
+  for (const object_ref<XFile>& file : all_files_handles) {
+    std::filesystem::path file_path =
+        std::filesystem::path(file->entry()->absolute_path());
+
+    auto symlinked_path = file_path.parent_path().string() + '\\';
+    if (symlinked_path == resolved_path) {
+      opened_files.push_back(file);
```

Comment:
It's not very desirable to create another vector overall, since that involves dynamic memory allocation, and even less to do that to just filter another vector. Currently, as there's only one usage of this function, it would be enough to just put that loop in CloseContent with ReleaseHandle done immediately. Later, I'm not sure, though a vector looks pretty clean from the readability perspective, something like a callback would have less unavoidable (on the usage side) overhead. Though it's just my personal opinion, other contributors may have a different preference.

Reply:
Implementation directly inside ContentClose was my scratch one, but it was quite ugly imo. 
Yes it adds overhead, but in this case should we even care? This vector shouldn't be freed after successful execution? 🤔

Also this method is used like max 20 times per game run (usually 2-3 times), so it is not that problematic in case of memory usage

Reply:
Heap allocations take a somewhat long time, though of course it's tiny relatively to file system calls, but still it's better to do avoid them where possible (though we have lots of std::vectors and long enough std::strings not to work with small string optimization in the emulator overall, so… ugh).